arrays/arrays.c

```diff
@@ -19,45 +19,56 @@ typedef struct Array {
 /*****
  * Allocate memory for a new array
  *****/
-Array *create_array (int capacity) {
+Array *create_array(int capacity)
+{
   // Allocate memory for the Array struct
-
+  Array *newarray = malloc(sizeof(Array));
   // Set initial values for capacity and count
-
+  newarray->capacity = capacity;
+  newarray->count = 0;
   // Allocate memory for elements
+  newarray->elements = calloc(capacity, sizeof(char *));
 
+  return newarray;
 }
 
-
 /*****
  * Free memory for an array and all of its stored elements
  *****/
-void destroy_array(Array *arr) {
+void destroy_array(Array *arr)
+{
 
   // Free all elements
-
+  free(arr->elements);
   // Free array
-
+  free(arr);
 }
 
 /*****
  * Create a new elements array with double capacity and copy elements
  * from old to new
  *****/
-void resize_array(Array *arr) {
+void *resize_array(Array *arr)
+{
 
   // Create a new element storage with double capacity
-
+  int capacity = arr->capacity;
+  char **newelements = calloc(capacity * 2, sizeof(char *));
   // Copy elements into the new storage
-
+  for (int i = 0; i < capacity; i++)
+  {
+    newelements[i] = arr->elements[i];
+    if (newarray->elements[i] != NULL)
+    {
+      newarray->count++;
+    }
+  }
   // Free the old elements array (but NOT the strings they point to)
-
+  destroy_array(arr);
   // Update the elements and capacity to new values
-
+  return newarray;
```

Comment:
Same idea, no need to return anything. This is supposed to be a void function, you're just changing the passed in array. You're not setting anything outside of the function.